styx-api-service/src/main/java/com/spotify/styx/api/StatusResource.java

```diff
@@ -73,14 +75,41 @@ public StatusResource(Storage storage) {
         Route.with(
             em.serializerDirect(EventsPayload.class),
             "GET", BASE + "/events/<cid>/<wfid>/<iid>",
-            rc -> eventsForWorkflowInstance(arg("cid", rc), arg("wfid", rc), arg("iid", rc))))
+            rc -> eventsForWorkflowInstance(arg("cid", rc), arg("wfid", rc), arg("iid", rc))),
+        Route.with(
+            em.response(TestServiceAccountUsageAuthorizationRequest.class,
+                TestServiceAccountUsageAuthorizationResponse.class),
+            "POST", BASE + "/testServiceAccountUsageAuthorization",
+            rc -> this::testServiceAccountUsageAuthorization)
+    )
 
         .map(r -> r.withMiddleware(Middleware::syncToAsync))
         .collect(toList());
 
     return Api.prefixRoutes(routes, V3);
   }
 
+  private Response<TestServiceAccountUsageAuthorizationResponse> testServiceAccountUsageAuthorization(
+      TestServiceAccountUsageAuthorizationRequest request) {
+    final Either<Response<?>, ServiceAccountUsageAuthorizer.ServiceAccountUsageAuthorizationResult> maybeResult =
+        accountUsageAuthorizer.authorizeServiceAccountUsage(request.serviceAccount(), request.principal());
+
+    if (maybeResult.isLeft()) {
+      throw new ResponseException(maybeResult.left().get());
```

Comment:
It seems we should return `maybeResult.left().get()` as it is, than throwing a runtime exception which will be interpreted as internal server error.

Reply:
This should be handled already by Middleware, not sure how it ended up with internal error.

Reply:
OK we are not attaching the right middleware. Fixed.